Automattic/newspack-popups
# fix(ga): add custom dimensions via Newspack instead of remote config …

### All Submissions:






### Changes proposed in this Pull Request:



Changes how segmentation-derived custom dimensions are supplied to GA (introduced in https://github.com/Automattic/newspack-popups/pull/325). 

The previous solution was to disable Site Kit's Analytics module's initial pageview reporting, in order to deliver the pageview to GA with this plugin, which appended segmentation-derived custom dimensions. This was error-prone, because if the site used something else on top of SiteKit to report to GA, this plugin would duplicate the pageview. 

The proposed solution uses Newspack's custom dimensions' values filtering to append the values before the page is sent to the client.  

### How to test the changes in this Pull Request:

1. Switch to `feat/custom-dimensions-filter` branch of Newspack Plugin (https://github.com/Automattic/newspack-plugin/pull/2148)
2. In Analytics wizard -> Custom Dimensions, configure two dimensions (create and assign a role in the table): one for Word count, one for Reader frequency (this one is segmentation-derived)
3. Visit a page, observe* that both custom dimensions are sent to GA** 

\* in network tab of devtools, filter by `collect` and look for a `pageview`
\** these are the `cd<index>` params in the request URL  

### Other information:
